### PR TITLE
update mm-sink tutorial and plot

### PR DESCRIPTION
**Pull request by zoepiran**
Updates MM-OT tutorial visualization and adds MM-OT plotting utils.

Updates in files:
- `docs/references.bib:` add citation `lin:22`
- `docs/tutorials/linear/600_mmsink.ipynb`: update visuals and add comparison to MM (det. solution)
- `src/ott/tools/plot.py`: add class `PlotMM`

### CODE REVIEW

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:324]: I'd add this option to `__init__`, as this will fail if there are more than 7 marginals (unlikely, but still).

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:314]: Would use `n_s = ot.shape`.

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:385]: As mentioned above, if possible, I would avoid iterating over the points.

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:320]: I'd consider passing `top_k` instead of saving it. Then in the notebook, you can use the `__call__` method directly for static plotting as well, instead of re-defining the functions.

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:407]: Would not save this, instead pass it to `__call__/update`.

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:381]: Please add a comment explaining what this does.

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:330]: Maybe instead of the hard-coded `0.6`, let's use `self._alpha` (also in other places)?

File: src/ott/tools/plot.py
Comment (michalk8) [inline src/ott/tools/plot.py:310]: As discussed below, don't think we need to store this.

File: src/ott/tools/plot.py
Comment (michalk8) [reply src/ott/tools/plot.py:407]: Also, why `top_k` should be by default `self._n`?

**Review by michalk8 — CHANGES_REQUESTED** [review]

File: docs/references.bib
Comment (zoepiran) [reply docs/references.bib:1020]: the JMLR citation doesnt include it, we can manually add:
`url={https://www.jmlr.org/papers/volume23/19-843/19-843.pdf},`
should I?

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:31]: to save large obj. but not necessary indeed.

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:367]: these are outputs of `mmsinkhorn.MMSinkhornOutput`, they do not follow same API.

Comment (zoepiran) [reply src/ott/tools/plot.py:367]: 🙏🏾

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:404]: 🙏🏾

Comment (zoepiran) [reply src/ott/tools/plot.py:404]: removed

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:405]: removed

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:315]: removed

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:368]: removed

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:392]: no; `ccworder` is only for sorting. 
because i wanted to fix the limits and we have normalized data. changed a bit.

Comment (zoepiran) [reply src/ott/tools/plot.py:392]: i think the change is ugly :(

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:342]: its the pain in the update indeed--can you figure the enumeration easily?

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:328]: Not sure i follow -> you need to extract from the indices object `top_k` vals

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:324]: i did something ugly; wdyt?

```
marker=markers[i % len(markers)],
c="black" if i < len(markers) else "grey",
```

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:314]: modified

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:385]: not sure how. will think of it

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:407]: because this is the minimal set of mapping, for each point single polygon

File: src/ott/tools/plot.py
Comment (zoepiran) [reply src/ott/tools/plot.py:310]: removed

**Comment by zoepiran** [comment]
<p>genious</p>

---

 <a href='https://app.reviewnb.com/ott-jax/ott/pull/572/discussion/#comment-2296263401'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by zoepiran** [comment]
<p>horrible phrasing</p>

---

 <a href='https://app.reviewnb.com/ott-jax/ott/pull/572/discussion/#comment-2296263414'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: src/ott/tools/plot.py
Comment (michalk8) [reply src/ott/tools/plot.py:343]: Will take a look!

**Review by michalk8 — APPROVED** [review]
LGTM, will wait until the CI passes and then merge! Thanks @zoepiran !

**Pull request merged**
